duration run_time
float64 route
